pkg/mappings.bzl

```diff
@@ -591,3 +611,117 @@ pkg_filegroup = rule(
     },
     provides = [PackageFilegroupInfo],
 )
+
+def _filter_directory_argify_pair(pair):
+    return "{}={}".format(*pair)
+
+def _filter_directory_impl(ctx):
+    out_dir = ctx.actions.declare_directory(ctx.attr.outdir_name or ctx.attr.name)
+
+    if not ctx.file.src.is_directory:
+        fail("Must be a directory (TreeArtifact)", "src")
+
+    args = ctx.actions.args()
+
+    # Flags
+    args.add_all(ctx.attr.excludes, before_each = "--exclude")
+    args.add_all(ctx.attr.renames.items(), before_each = "--rename", map_each = _filter_directory_argify_pair)
+
+    args.add("--prefix", ctx.attr.prefix)
+    args.add("--strip_prefix", ctx.attr.strip_prefix)
+
+    # Adding the directories directly here requires manually specifying the
+    # path.  Bazel will reject simply passing in the File object.
+    args.add(ctx.file.src.path)
+    args.add(out_dir.path)
+
+    ctx.actions.run(
+        executable = ctx.executable._filterer,
+        use_default_shell_env = True,
+        arguments = [args],
+        inputs = [ctx.file.src],
+        outputs = [out_dir],
+    )
+
+    return [DefaultInfo(files = depset([out_dir]))]
+
+filter_directory = rule(
+    doc = """
+    Transform directories (TreeArtifacts) using pkg_filegroup-like semantics.
```

Comment:
I think the doc text has to start on the """ line

Reply:
Done.